File: src/main/kotlin/com/kursx/parser/fb2/Title.java
package com.kursx.parser.fb2;

import org.jetbrains.annotations.NotNull;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import java.util.ArrayList;

@SuppressWarnings("unused")
public class Title {

    protected ArrayList<P> paragraphs = new ArrayList<>();

    public Title() {
    }

    Title(Node root) {
        NodeList body = root.getChildNodes();
        for (int item = 0; item < body.getLength(); item++) {
            Node node = body.item(item);
            if ("p".equals(node.getNodeName())) {
                paragraphs.add(new P(node));
            }
        }
    }

    @NotNull
    public ArrayList<P> getParagraphs() {
        return paragraphs;
    }
}

File: src/main/kotlin/com/kursx/parser/fb2/Title.kt
package com.kursx.parser.fb2

import kotlinx.serialization.Serializable
import org.w3c.dom.Node

@Suppress("unused")
@Serializable
class Title {
    var paragraphs = ArrayList<P>()
        protected set

    constructor()
    internal constructor(root: Node) {
        val body = root.childNodes
        for (item in 0 until body.length) {
            val node = body.item(item)
            if ("p" == node.nodeName) {
                paragraphs.add(P(node))
            }
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Title

        return paragraphs == other.paragraphs
    }

    override fun hashCode(): Int {
        return paragraphs.hashCode()
    }

}
